cmake/app/boilerplate.cmake

```diff
@@ -181,6 +181,13 @@ endif()
 assert(BOARD "BOARD not set")
 message(STATUS "Board: ${BOARD}")
 
+include(${ZEPHYR_BASE}/boards/deprecated.cmake)
+if(${BOARD}_DEPRECATED)
+  set(BOARD_DEPRECATED ${BOARD})
+  set(BOARD ${${BOARD}_DEPRECATED})
+  message(WARNING "Deprecated BOARD=${BOARD_DEPRECATED} name specified, board automatically changed to: ${BOARD}.")
+endif()
+
```

Comment:
This now produces an unwanted warning about pristinely cleaning the build dir when we do this:

```
sebo@ubuntu:~/ncs/zephyr/samples/hello_world$ cmake -H. -Bb -DBOARD=nrf51_pca10028
-- Application: /home/sebo/ncs/zephyr/samples/hello_world
-- Zephyr version: 2.2.99
-- Found PythonInterp: /usr/bin/python3.6 (found suitable version "3.6.9", minimum required is "3.6") 
-- Board: nrf51_pca10028
CMake Warning at /home/sebo/ncs/zephyr/cmake/app/boilerplate.cmake:188 (message):
  Deprecated BOARD=nrf51_pca10028 name specified, board automatically changed
  to: nrf51dk_nrf51422.
Call Stack (most recent call first):
  CMakeLists.txt:5 (include)


-- Found west: /home/sebo/.local/bin/west (found suitable version "0.6.3", minimum required is "0.6.0")
-- Found toolchain: zephyr (/home/sebo/zephyr-sdk-0.11.2)
-- Loading /home/sebo/ncs/zephyr/boards/arm/nrf51dk_nrf51422/nrf51dk_nrf51422.dts as base
Devicetree header saved to '/home/sebo/ncs/zephyr/samples/hello_world/b/zephyr/include/generated/devicetree_unfixed.h'
Parsing /home/sebo/ncs/zephyr/Kconfig
Loaded configuration '/home/sebo/ncs/zephyr/boards/arm/nrf51dk_nrf51422/nrf51dk_nrf51422_defconfig'
Merged configuration '/home/sebo/ncs/zephyr/samples/hello_world/prj.conf'
Configuration saved to '/home/sebo/ncs/zephyr/samples/hello_world/b/zephyr/.config'
Kconfig header saved to '/home/sebo/ncs/zephyr/samples/hello_world/b/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 9.2.0
-- The CXX compiler identification is GNU 9.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/sebo/zephyr-sdk-0.11.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Cache files will be written to: /home/sebo/.cache/zephyr
-- Configuring done
-- Generating done
-- Build files have been written to: /home/sebo/ncs/zephyr/samples/hello_world/b
sebo@ubuntu:~/ncs/zephyr/samples/hello_world$ touch CMakeLists.txt 
sebo@ubuntu:~/ncs/zephyr/samples/hello_world$ ninja -C b/
ninja: Entering directory `b/'
[0/1] Re-running CMake...
-- Application: /home/sebo/ncs/zephyr/samples/hello_world
-- Zephyr version: 2.2.99
CMake Warning at /home/sebo/ncs/zephyr/cmake/app/boilerplate.cmake:162 (message):
  The build directory must be cleaned pristinely when changing boards
Call Stack (most recent call first):
  CMakeLists.txt:5 (include)


-- Board: nrf51dk_nrf51422
-- Found west: /home/sebo/.local/bin/west (found suitable version "0.6.3", minimum required is "0.6.0")
-- Found toolchain: zephyr (/home/sebo/zephyr-sdk-0.11.2)
-- Loading /home/sebo/ncs/zephyr/boards/arm/nrf51dk_nrf51422/nrf51dk_nrf51422.dts as base
Devicetree header saved to '/home/sebo/ncs/zephyr/samples/hello_world/b/zephyr/include/generated/devicetree_unfixed.h'
Parsing /home/sebo/ncs/zephyr/Kconfig
Loaded configuration '/home/sebo/ncs/zephyr/samples/hello_world/b/zephyr/.config'
No change to configuration in '/home/sebo/ncs/zephyr/samples/hello_world/b/zephyr/.config'
No change to Kconfig header in '/home/sebo/ncs/zephyr/samples/hello_world/b/zephyr/include/generated/autoconf.h'
-- Cache files will be written to: /home/sebo/.cache/zephyr
-- Configuring done
-- Generating done
-- Build files have been written to: /home/sebo/ncs/zephyr/samples/hello_world/b

```

Reply:
I can't figure out how to fix this, so I think we should just live with it.

Reply:
interesting, cause the cached board should be the switched one, ie `nrf51dk_nrf51422`.

Will tru to look into a fix, cause this is not nice to users.

Thanks foir spotting.

Reply:
Fixed.
Please recheck, and see if the behavior is as you like.